flink/v1.16/flink/src/main/java/org/apache/iceberg/flink/source/FlinkInputFormat.java

```diff
@@ -79,15 +68,9 @@ public BaseStatistics getStatistics(BaseStatistics cachedStatistics) {
   @Override
   public FlinkInputSplit[] createInputSplits(int minNumSplits) throws IOException {
     // Called in Job manager, so it is OK to load table from catalog.
-    tableLoader.open();
     final ExecutorService workerPool =
         ThreadPools.newWorkerPool("iceberg-plan-worker-pool", context.planParallelism());
-    try (TableLoader loader = tableLoader) {
-      Table table = loader.loadTable();
-      return FlinkSplitPlanner.planInputSplits(table, context, workerPool);
-    } finally {
-      workerPool.shutdown();
-    }
+    return FlinkSplitPlanner.planInputSplits(table, context, workerPool);
```

Comment:
just to double check. input format is only for batch query, right? SerializableTable is read only and can't be used for long-running streaming source.

Reply:
Flink source uses the monitor function to monitor snapshots and forward splits to `StreamingReaderOperator`, `StreamingReaderOperator` uses `FlinkInputFormat` to open splits. Opening splits should work even the table is readonly.

Reply:
yeah. using readonly table for reader function is fine.